docs/log_format.md

```diff
@@ -0,0 +1,58 @@
+# Log Format
+
+## Log Line Prefix
+
+Shadow produces simulator log messages in the following format:
+
+```text
+real-time [thread-id] virtual-time [logdomain-loglevel] [hostname~ip] [function-name] MESSAGE
+```
+
+**FIXME**: The `real-time` and `virtual-time` use different time units. Is this correct, or should we change this in Shadow?
```

Comment:
You mean that real-time shows microsecond granularity, while virtual-time shows nano-second granularity? Yeah, I noticed and preserved this when porting the logger. I don't think finer granularity than microseconds for real-time is likely to be useful; and it probably makes sense to log sim time at nanosecond granularity since that's the internal granularity.

I'm inclined to leave as-is, though could be convinced to extend real-time to nanoseconds if we think it'd be helpful

Alternatively maybe it'd be clearer if we changed the separator after seconds to be `.`, so that the format of both could be expressed as `hours:minutes:seconds`? (Both fields should already be left-padded; e.g. 1 microsecond is 00:00:00:000001 not 00:00:00:1)

Reply:
Actually the separator after seconds *is* `.`, not `:`

Reply:
Oops, I meant to look into this before adding the PR. If it uses a `.` separator and is left-padded, then the current behaviour sounds good to me. I'll update the docs to just say `hours:minutes:seconds` for both.